Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m

```diff
@@ -275,42 +273,48 @@ - (CLLocationManager*) locationManager
 - (void)locationManager:(CLLocationManager *) __unused manager
      didUpdateLocations:(NSArray *)locations
 {
+
     CLLocation * currentLocation = [locations lastObject];
     CLLocationCoordinate2D coordinate = [currentLocation coordinate];
     
     float lat = coordinate.latitude;
     float lon = coordinate.longitude;
     
-    NSTimeInterval lastAQICheckedTime = [[NSUserDefaults standardUserDefaults] doubleForKey:kAQILastChecked];
-    NSTimeInterval currentTime = [[NSDate date] timeIntervalSinceReferenceDate];
+    NSDate *lastAQICheckedTime = [[NSUserDefaults standardUserDefaults] objectForKey:kAQILastChecked] ?: nil;
+    
+    NSDate *currentTime = [NSDate date];
     self.aqiResponse = [[NSMutableDictionary alloc]init];
     
     __weak APHAirQualityDataModel *weakSelf = self;
     
-    if (!self.fetchingAirQualityReport && ((currentTime - lastAQICheckedTime) >= kAQICheckInterval)) {
+    if (!self.fetchingAirQualityReport && (([currentTime timeIntervalSinceDate: lastAQICheckedTime]) >= kAQICheckInterval)) {
+        [[NSUserDefaults standardUserDefaults] setObject:[NSDate new] forKey:kAQILastChecked];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+        
+        [self createZipArchive];
         self.fetchingAirQualityReport = YES;
         //Make Network call for Air Quality in block
-            [self.networkManager get:kAlertGetJson headers:nil parameters:@{@"lat":@(lat),@"lon":@(lon)} completion:^(NSURLSessionDataTask __unused *task, id responseObject, NSError *error) {
-                APCLogError2(error);
-                weakSelf.fetchingAirQualityReport = NO;
-                if (!error) {
-                    weakSelf.aqiObject.aqiDictionary = responseObject;
-                    weakSelf.aqiResponse = [[NSMutableDictionary alloc]initWithDictionary:responseObject];
-                    
-                    //send notification for [APHDashboardViewController to prepareData];
-                    APCLogEventWithData(kNetworkEvent, (@{
-                                                          @"event_detail" : @"Received Air Quality Info from Server"
-                                                          }));
-                    
-                    //sends a new AQIAlert to the dashboard. The dashboard should receive it on the main thread.
-                    [weakSelf.airQualityReportReceiver airQualityModel:weakSelf didDeliverAirQualityAlert:weakSelf.aqiObject];
+        [self.networkManager get:kAlertGetJson headers:nil parameters:@{@"lat":@(lat),@"lon":@(lon)} completion:^(NSURLSessionDataTask __unused *task, id responseObject, NSError *error) {
+            APCLogError2(error);
+            weakSelf.fetchingAirQualityReport = NO;
+            if (!error) {
```

Comment:
In block use ‘strongSelf’ retains it, and it stays retained until the block returns.
```
// In block
__typeof__(self) strongSelf = weakSelf;
[strongSelf doSomething];
```